ann_benchmarks/datasets.py

```diff
@@ -219,11 +220,25 @@ def word2bits(out_fn, path, fn):
         n_words, k = [int(z) for z in next(f).strip().split()]
         X = numpy.zeros((n_words, k), dtype=numpy.bool)
         for i in range(n_words):
-            X[i] = [float(z) > 0 for z in next(f).strip().split()[1:]]
+            X[i] = numpy.array([float(z) > 0 for z in next(f).strip().split()[1:]], dtype=numpy.bool)
 
-        X_train, X_test = train_test_split(X)
-        write_output(X_train, X_test, out_fn, 'euclidean')  # TODO: use hamming
+        X_train, X_test = train_test_split(X, test_size=1000)
+        write_output(X_train, X_test, out_fn, 'hamming', 'bit')
 
+def sift_hamming(out_fn, fn):
+    import tarfile
+    local_fn = fn + '.tar.gz'
+    url = 'http://sss.projects.itu.dk/ann-benchmarks/datasets/%s.tar.gz' % fn
```

Comment:
See above: 

> I have added a binary version of the SIFT dataset to it, generated using Spherical hashing. Unfortunately, I don't know of a nice Python implementation of Spherical hashing which would allow producing the binary embedding from SIFT directly from the source (and for other of our datasets as well).

It's [this method](http://www.ee.columbia.edu/ln/dvmm/pubs/files/Spherical_Hashing.pdf) using their implementation from https://github.com/jp-sglab/Spherical_Hashing.